packages/plugins/test/mocha/simulate.spec.ts

```diff
@@ -225,3 +236,186 @@ describe('handleSimulationSuccess', () => {
     }
   })
 })
+
+describe('createHardhatEthersProviderProxy', () => {
```

Comment:
Nit: It's confusing to me to have tests in `simulate.spec.ts` that don't interact with live networks and run on all PRs. I understand why you included them here since these are logic tests related to the simulation. Still, it would be slightly better to keep the tests that run against live network RPC providers separated into their own file for the sake of organization. I would use a different name for this file, like `live-network.spec.ts`. I would then have a `simulation.spec.ts` file that only includes these new tests.

Reply:
Will keep this unchanged for the sake of speed, but I'll make a ticket to consider re-organizing this test file. Fwiw, our test suite generally follows the pattern of using the same name for the test file and the file being tested. In this case, `createHardhatEthersProviderProxy` is in `simulate.ts`, which is why I put this test suite in `simulate.spec.ts`